### DISCUSSION TITLE
Error :: Add-PnPCustomAction : Attempted to perform an unauthorized operation

### DISCUSSION THREAD

**Discussion by MAP-Git** [pnp/powershell#2369] [Q&A]
Getting this error: "Add-PnPCustomAction : Attempted to perform an unauthorized operation." User has Admin level access on SharePoint Online. PnP PowerShell throws error. Please advice solution here.

**Comment by fofawowe** [comment]
I am getting the same error running this command when using an azure app with certificate based auth. The app has api permissions "Sites.FullControl.All" and I'm using PnP.Powershell version 1.11.0.
Any ideas?

**Comment by rohit404404** [comment]
Can you try disabling  NoScript parameter?  For me that was the issue. 

`Set-PnPSite -NoScriptSite $false`

**Reply by NewGHUser4321** [reply]
This worked for me too. I had to do this on root Site Collection. On other sites, it worked fine without having to disable the NoScript Parameter.
